internal/controller/vrg_volsync.go

```diff
@@ -640,13 +673,43 @@ func protectedPVCAnnotations(pvc corev1.PersistentVolumeClaim) map[string]string
 	return res
 }
 
+func (v *VRGInstance) pvcUnprotectVolSyncIfDeleted(
+	pvc corev1.PersistentVolumeClaim, log logr.Logger,
+) (pvcDeleted bool) {
+	pvcDeleted = util.ResourceIsDeleted(&pvc)
+	if !pvcDeleted {
+		return
+	}
+
+	log.Info("PVC unprotect VolSync", "deletion time", pvc.GetDeletionTimestamp())
+	v.pvcUnprotectVolSync(pvc, log)
+
+	return
+}
+
+func (v *VRGInstance) protectedByVolsync(pvc *corev1.PersistentVolumeClaim) bool {
+	for _, vsPVC := range v.volSyncPVCs {
+		if vsPVC.GetName() == pvc.GetName() &&
+			vsPVC.GetNamespace() == pvc.GetNamespace() {
+			return true
+		}
+	}
+
+	return false
+}
+
 func (v *VRGInstance) pvcUnprotectVolSync(pvc corev1.PersistentVolumeClaim, log logr.Logger) {
-	if !VolumeUnprotectionEnabledForAsyncVolSync {
-		log.Info("Volume unprotection disabled for VolSync")
+	if !v.protectedByVolsync(&pvc) {
```

Comment:
When called from `pvcsDeselectedUnprotect` the PVC to unprotect will not be in v.volSyncPVCs, so the check `protectedByVolsync` will always be false, and we will skip further processing?

Testing showed that deselected PVCs are correctly removed form protection, so not sure what I am missing.

Reply:
If it is deleted, it will be there, but if the label is removed, then yes, it won't be there. So maybe we listByOwner instead, and in that case, we need to add a volsync owner.